src/containers/Auth/SignUp/SignUp.js

```diff
@@ -0,0 +1,153 @@
+
+    return (
+      <div className="sign-margin">
+        {authRedirectPath}
+        <Grid centered>
+          <Grid.Column style={{ maxWidth: 550, marginTop: 20 }}>
+            <Header style={{ color: "#4183c4" }} as="h2">
+              Sign Up
+            </Header>
+            <Form size="big">
+            {formInput({name:'firstname',
+                label:"First Name",
+                value:firstname,
+                placeholder:'firstname',
+                onChange:this.onChange})}
+
+            {formInput({name:'lastname',
+                label:"Last Name",
+                value:lastname,
+                placeholder:'lastname',
+                onChange:this.onChange})}
+
+             {formInput({name:'email',
+                label:"Email Address",
+                value:email,
+                placeholder:'email',
+                onChange:this.onChange})}
+              {error ? <p style={{ color: "red", fontSize: 13 }}>{error}</p> : errorProp==="The email address is already in use by another account." ?  <p style={{ color: "red", fontSize: 13 }}>{errorProp}</p> : ""}
+
+             {formInput({name:'password',
+                type:"password",
+                label:"Password",
+                value:password,
+                placeholder:'password',
+                onChange:this.onChange})}
+              <Form.Field>
+                {radioButton({ label:"seller",
+                  name:"radioGroup",
+                  value:"seller",
+                  usertype:"seller",
+                  checked:this.state.usertype === "seller",
+                  onChange:this.handleChange
+                  })}
+              </Form.Field>
+              <Form.Field>
+              {radioButton({ label:"customer",
+                  name:"radioGroup",
+                  value:"customer",
+                  usertype:"customer",
+                  checked:this.state.usertype === "customer",
+                  onChange:this.handleChange
+                  })}
+              </Form.Field>
+              <Button
+                primary
+                disabled={isInvalid}
+                control={Button}
+                onClick={this.onSubmit}
+              >
+                Submit
+              </Button>
+            </Form>
+            {/* <SignInLink/> */}
+            <p>
+              Already have an account? <Link to="/signin">Sign In</Link>
+            </p>
+          </Grid.Column>
+        </Grid>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = (state) => {
+  return {
+    errorProp: state.userReducer.error,
+    loading: state.userReducer.loading,
+    isLoginSuccess: state.userReducer.isLoginSuccess,
+    redirectPath: state.userReducer.authRedirectPath,
+  };
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    onRegister: (userinfo) => dispatch(registerUser(userinfo)),
+  };
+};
+
+export default formHoc(connect(mapStateToProps, mapDispatchToProps)(SignUpFormBase));
```

Comment:
Form validation can be managed from HOC component instead of manage it separately from individual components.